Compiler/src/Compiler.cpp

```diff
@@ -3703,15 +3701,23 @@ void compileOrThrow(BytecodeBuilder& bytecode, AstStatBlock* root, const AstName
 
     Compiler compiler(bytecode, options);
 
-    // since access to some global objects may result in values that change over time, we block table imports
-    for (const char* global : kSpecialGlobals)
+    // since access to some global objects may result in values that change over time, we block imports from non-readonly tables
     {
-        AstName name = names.get(global);
+        AstName name = names.get("_G");
 
         if (name.value)
-            compiler.globals[name].special = true;
+            compiler.globals[name].writable = true;
     }
 
+    if (options.mutableGlobalNames)
+        for (const char** ptr = options.mutableGlobalNames; *ptr != NULL; ptr += sizeof(const char*))
```

Comment:
This code isn't correct -- it should be `++ptr`. Please add a test for this feature to Compiler.test.cpp, as that would make the issue obvious + provide test coverage.